cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -482,10 +481,8 @@ public Void call() throws Exception {
           context.setState(new ProgramState(ProgramStatus.INITIALIZING, null));
           if (mapReduce instanceof ProgramLifecycle) {
```

Comment:
No. When user's MapReduce program directly implements pre 3.5 `MapReduce` interface, then it won't be true.